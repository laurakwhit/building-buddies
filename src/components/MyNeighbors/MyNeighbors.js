import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

import Nav from '../Nav/Nav';
import LoadingSpinner from '../LoadingSpinner/LoadingSpinner';

import './MyNeighbors.scss';

class MyNeighbors extends Component {
  async componentDidMount() {
    const { getNeighbors } = this.props;
    await getNeighbors();
  }

  displayNeighbors = () => {
    const { neighbors } = this.props;
    if (!neighbors) return <LoadingSpinner />;
    return neighbors.map((neighbor, i) => (
      <div className="neighbor" key={`${neighbor.name}-${i}`}>
        <Link to={`/neighbors/${neighbor.id}`}>{neighbor.name}</Link>
      </div>
    ));
  };

  render() {
    const displayNeighbors = this.displayNeighbors();

    return (
      <div className="my-neighbors">
        <Nav />
        <div className="main-content">{displayNeighbors}</div>
      </div>
    );
  }
}

MyNeighbors.propTypes = {
  userBuilding: PropTypes.object,
  neighbors: PropTypes.array
};

export default MyNeighbors;
