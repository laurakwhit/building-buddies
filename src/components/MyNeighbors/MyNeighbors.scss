.my-neighbors {
  display: flex;
  .main-content {
    overflow: scroll;
    height: 100vh;
    width: calc(100% - 20rem);
    .name-link {
      text-decoration: none;
    }
    .neighbor {
      display: flex;
      justify-content: center;
      align-items: center;
      height: 3rem;
      width: 55%;
      margin: 2rem auto;
      border-radius: 10px;
      background: #0f2027; /* fallback for old browsers */
      background: -webkit-linear-gradient(
        to right,
        #2c5364,
        #203a43,
        #0f2027
      ); /* Chrome 10-25, Safari 5.1-6 */
      background: linear-gradient(
        to right,
        #2c5364,
        #203a43,
        #0f2027
      ); /* W3C, IE 10+/ Edge, Firefox 16+, Chrome 26+, Opera 12+, Safari 7+ */

      -webkit-box-shadow: 11px 10px 36px -12px rgba(0, 0, 0, 0.66);
      -moz-box-shadow: 11px 10px 36px -12px rgba(0, 0, 0, 0.66);
      box-shadow: 11px 10px 36px -12px rgba(0, 0, 0, 0.66);
      color: #fff;
      font-size: 2rem;
      .fas {
        padding-left: 2rem;
      }
      &:hover {
        transform: scale(1.1);
        transition: ease-in-out 0.2s;
      }
    }
  }
}
