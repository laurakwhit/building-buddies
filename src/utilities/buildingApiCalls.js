export const getAllBuildings = async () => {
  const response = await fetch(`${process.env.REACT_APP_API}/api/v1/buildings`);
  const buildings = await response.json();
  return buildings;
};

export const getBuilding = async () => {};

export const getAllBuildingUsers = async buildingId => {
  const response = await fetch(
    `${process.env.REACT_APP_API}/api/v1/buildings/${buildingId}/users`
  );
  const buildingUsers = await response.json();
  return buildingUsers;
};
