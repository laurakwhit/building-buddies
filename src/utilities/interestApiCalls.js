export const getAllInterests = async () => {
  const response = await fetch(`${process.env.REACT_APP_API}/api/v1/interests`);
  const interests = await response.json();
  return interests;
};

export const addInterest = async () => {};
