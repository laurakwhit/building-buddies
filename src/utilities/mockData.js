export const mockInterests = [{ "id": 1, "name": "golf", "created_at": "2018-10-10T19:53:00.107Z", "updated_at": "2018-10-10T19:53:00.107Z" }, { "id": 2, "name": "gardening", "created_at": "2018-10-10T19:53:00.107Z", "updated_at": "2018-10-10T19:53:00.107Z" }, { "id": 3, "name": "camping", "created_at": "2018-10-10T19:53:00.107Z", "updated_at": "2018-10-10T19:53:00.107Z" }, { "id": 4, "name": "rock climbing", "created_at": "2018-10-10T19:53:00.107Z", "updated_at": "2018-10-10T19:53:00.107Z" }, { "id": 5, "name": "fly fishing", "created_at": "2018-10-10T19:53:00.107Z", "updated_at": "2018-10-10T19:53:00.107Z" }, { "id": 6, "name": "coding", "created_at": "2018-10-10T19:53:00.107Z", "updated_at": "2018-10-10T19:53:00.107Z" }, { "id": 7, "name": "working on cars", "created_at": "2018-10-10T19:53:00.107Z", "updated_at": "2018-10-10T19:53:00.107Z" }, { "id": 8, "name": "bowling", "created_at": "2018-10-10T19:53:00.107Z", "updated_at": "2018-10-10T19:53:00.107Z" }, { "id": 9, "name": "cooking", "created_at": "2018-10-10T19:53:00.107Z", "updated_at": "2018-10-10T19:53:00.107Z" }, { "id": 10, "name": "hiking", "created_at": "2018-10-10T19:53:00.107Z", "updated_at": "2018-10-10T19:53:00.107Z" }, { "id": 11, "name": "skiing", "created_at": "2018-10-10T19:53:00.107Z", "updated_at": "2018-10-10T19:53:00.107Z" }, { "id": 12, "name": "snowboarding", "created_at": "2018-10-10T19:53:00.107Z", "updated_at": "2018-10-10T19:53:00.107Z" }, { "id": 13, "name": "mountain biking", "created_at": "2018-10-10T19:53:00.107Z", "updated_at": "2018-10-10T19:53:00.107Z" }, { "id": 14, "name": "knitting", "created_at": "2018-10-10T19:53:00.107Z", "updated_at": "2018-10-10T19:53:00.107Z" }]
export const mockUser = {
  building_id: 291,
  email: "asdf@asdf.com",
  name: "asdf"
};