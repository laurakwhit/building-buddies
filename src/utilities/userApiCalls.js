export const getUserInterests = async user_id => {
  const response = await fetch(
    `${process.env.REACT_APP_API}/api/v1/users/${user_id}/interests`
  );
  const interests = await response.json();
  return interests;
};

export const addUser = async ({ name, email, password, building_id }) => {
  const response = await fetch(`${process.env.REACT_APP_API}/api/v1/users`, {
    method: 'POST',
    body: JSON.stringify({ name, email, password, building_id }),
    headers: {
      'Content-Type': 'application/json'
    }
  });
  const data = await response.json();
  return data;
};

export const addUserInterest = async ({ user_id, interest_id }) => {
  const response = await fetch(
    `${
      process.env.REACT_APP_API
    }/api/v1/users/${user_id}/interests/${interest_id}`,
    {
      method: 'POST'
    }
  );
  const data = await response.json();
  return data;
};

export const deleteUserInterest = async ({ user_id, interest_id }) => {
  const response = await fetch(
    `${
      process.env.REACT_APP_API
    }/api/v1/users/${user_id}/interests/${interest_id}`,
    {
      method: 'DELETE',
      body: JSON.stringify({ interest_id }),
      headers: {
        'Content-Type': 'application/json'
      }
    }
  );
  const data = await response.json();
  return data;
};
