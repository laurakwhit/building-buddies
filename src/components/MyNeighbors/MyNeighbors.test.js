import React from 'react';
import { shallow } from 'enzyme';
import { mockNeighbors } from '../../utilities/mockData';
import MyNeighbors from './MyNeighbors';

describe('MyNeighbors component', () => {
  let wrapper;
  let mockGetNeighbors;

  beforeEach(() => {
    mockGetNeighbors = jest.fn();
    wrapper = shallow(
      <MyNeighbors getNeighbors={mockGetNeighbors} neighbors={mockNeighbors} />
    );
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
