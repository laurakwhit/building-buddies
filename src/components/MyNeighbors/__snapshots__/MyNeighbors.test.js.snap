// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyNeighbors component should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MyNeighbors
    getNeighbors={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    neighbors={
      Array [
        Object {
          "building_id": 1,
          "created_at": "2018-10-14T18:06:35.045Z",
          "email": "gray.smith@gmail.com",
          "id": 1,
          "name": "Gray Smith",
          "password": "asdfasdf",
          "updated_at": "2018-10-14T18:06:35.045Z",
        },
        Object {
          "building_id": 1,
          "created_at": "2018-10-14T18:06:35.045Z",
          "email": "boolean.brandon@gmail.com",
          "id": 4,
          "name": "Boolean Brandon",
          "password": "asdfasdf",
          "updated_at": "2018-10-14T18:06:35.045Z",
        },
        Object {
          "building_id": 1,
          "created_at": "2018-10-14T18:06:35.045Z",
          "email": "tim.fishcher@gmail.com",
          "id": 7,
          "name": "Tim Fischer",
          "password": "asdfasdf",
          "updated_at": "2018-10-14T18:06:35.045Z",
        },
      ]
    }
    userBuilding={
      Object {
        "address": "1910 S Josephine St
Denver, CO
",
        "created_at": "2018-10-14T18:06:35.020Z",
        "id": 1,
        "name": "Modera Observatory Park",
        "updated_at": "2018-10-14T18:06:35.020Z",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Nav />,
        <div
          className="main-content"
        >
          <Link
            className="name-link"
            replace={false}
            to="/neighbors/1"
          >
            <div
              className="neighbor"
            >
              <h4>
                Gray Smith
              </h4>
              <i
                className="fas fa-angle-right"
              />
            </div>
          </Link>
          <Link
            className="name-link"
            replace={false}
            to="/neighbors/4"
          >
            <div
              className="neighbor"
            >
              <h4>
                Boolean Brandon
              </h4>
              <i
                className="fas fa-angle-right"
              />
            </div>
          </Link>
          <Link
            className="name-link"
            replace={false}
            to="/neighbors/7"
          >
            <div
              className="neighbor"
            >
              <h4>
                Tim Fischer
              </h4>
              <i
                className="fas fa-angle-right"
              />
            </div>
          </Link>
        </div>,
      ],
      "className": "my-neighbors",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Link
              className="name-link"
              replace={false}
              to="/neighbors/1"
            >
              <div
                className="neighbor"
              >
                <h4>
                  Gray Smith
                </h4>
                <i
                  className="fas fa-angle-right"
                />
              </div>
            </Link>,
            <Link
              className="name-link"
              replace={false}
              to="/neighbors/4"
            >
              <div
                className="neighbor"
              >
                <h4>
                  Boolean Brandon
                </h4>
                <i
                  className="fas fa-angle-right"
                />
              </div>
            </Link>,
            <Link
              className="name-link"
              replace={false}
              to="/neighbors/7"
            >
              <div
                className="neighbor"
              >
                <h4>
                  Tim Fischer
                </h4>
                <i
                  className="fas fa-angle-right"
                />
              </div>
            </Link>,
          ],
          "className": "main-content",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "Gray Smith-0",
            "nodeType": "class",
            "props": Object {
              "children": <div
                className="neighbor"
              >
                <h4>
                  Gray Smith
                </h4>
                <i
                  className="fas fa-angle-right"
                />
              </div>,
              "className": "name-link",
              "replace": false,
              "to": "/neighbors/1",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h4>
                    Gray Smith
                  </h4>,
                  <i
                    className="fas fa-angle-right"
                  />,
                ],
                "className": "neighbor",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Gray Smith",
                  },
                  "ref": null,
                  "rendered": "Gray Smith",
                  "type": "h4",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "fas fa-angle-right",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "i",
                },
              ],
              "type": "div",
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "Boolean Brandon-1",
            "nodeType": "class",
            "props": Object {
              "children": <div
                className="neighbor"
              >
                <h4>
                  Boolean Brandon
                </h4>
                <i
                  className="fas fa-angle-right"
                />
              </div>,
              "className": "name-link",
              "replace": false,
              "to": "/neighbors/4",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h4>
                    Boolean Brandon
                  </h4>,
                  <i
                    className="fas fa-angle-right"
                  />,
                ],
                "className": "neighbor",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Boolean Brandon",
                  },
                  "ref": null,
                  "rendered": "Boolean Brandon",
                  "type": "h4",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "fas fa-angle-right",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "i",
                },
              ],
              "type": "div",
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "Tim Fischer-2",
            "nodeType": "class",
            "props": Object {
              "children": <div
                className="neighbor"
              >
                <h4>
                  Tim Fischer
                </h4>
                <i
                  className="fas fa-angle-right"
                />
              </div>,
              "className": "name-link",
              "replace": false,
              "to": "/neighbors/7",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h4>
                    Tim Fischer
                  </h4>,
                  <i
                    className="fas fa-angle-right"
                  />,
                ],
                "className": "neighbor",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Tim Fischer",
                  },
                  "ref": null,
                  "rendered": "Tim Fischer",
                  "type": "h4",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "fas fa-angle-right",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "i",
                },
              ],
              "type": "div",
            },
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Nav />,
          <div
            className="main-content"
          >
            <Link
              className="name-link"
              replace={false}
              to="/neighbors/1"
            >
              <div
                className="neighbor"
              >
                <h4>
                  Gray Smith
                </h4>
                <i
                  className="fas fa-angle-right"
                />
              </div>
            </Link>
            <Link
              className="name-link"
              replace={false}
              to="/neighbors/4"
            >
              <div
                className="neighbor"
              >
                <h4>
                  Boolean Brandon
                </h4>
                <i
                  className="fas fa-angle-right"
                />
              </div>
            </Link>
            <Link
              className="name-link"
              replace={false}
              to="/neighbors/7"
            >
              <div
                className="neighbor"
              >
                <h4>
                  Tim Fischer
                </h4>
                <i
                  className="fas fa-angle-right"
                />
              </div>
            </Link>
          </div>,
        ],
        "className": "my-neighbors",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Link
                className="name-link"
                replace={false}
                to="/neighbors/1"
              >
                <div
                  className="neighbor"
                >
                  <h4>
                    Gray Smith
                  </h4>
                  <i
                    className="fas fa-angle-right"
                  />
                </div>
              </Link>,
              <Link
                className="name-link"
                replace={false}
                to="/neighbors/4"
              >
                <div
                  className="neighbor"
                >
                  <h4>
                    Boolean Brandon
                  </h4>
                  <i
                    className="fas fa-angle-right"
                  />
                </div>
              </Link>,
              <Link
                className="name-link"
                replace={false}
                to="/neighbors/7"
              >
                <div
                  className="neighbor"
                >
                  <h4>
                    Tim Fischer
                  </h4>
                  <i
                    className="fas fa-angle-right"
                  />
                </div>
              </Link>,
            ],
            "className": "main-content",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "Gray Smith-0",
              "nodeType": "class",
              "props": Object {
                "children": <div
                  className="neighbor"
                >
                  <h4>
                    Gray Smith
                  </h4>
                  <i
                    className="fas fa-angle-right"
                  />
                </div>,
                "className": "name-link",
                "replace": false,
                "to": "/neighbors/1",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <h4>
                      Gray Smith
                    </h4>,
                    <i
                      className="fas fa-angle-right"
                    />,
                  ],
                  "className": "neighbor",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Gray Smith",
                    },
                    "ref": null,
                    "rendered": "Gray Smith",
                    "type": "h4",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": "fas fa-angle-right",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "i",
                  },
                ],
                "type": "div",
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "Boolean Brandon-1",
              "nodeType": "class",
              "props": Object {
                "children": <div
                  className="neighbor"
                >
                  <h4>
                    Boolean Brandon
                  </h4>
                  <i
                    className="fas fa-angle-right"
                  />
                </div>,
                "className": "name-link",
                "replace": false,
                "to": "/neighbors/4",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <h4>
                      Boolean Brandon
                    </h4>,
                    <i
                      className="fas fa-angle-right"
                    />,
                  ],
                  "className": "neighbor",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Boolean Brandon",
                    },
                    "ref": null,
                    "rendered": "Boolean Brandon",
                    "type": "h4",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": "fas fa-angle-right",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "i",
                  },
                ],
                "type": "div",
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "Tim Fischer-2",
              "nodeType": "class",
              "props": Object {
                "children": <div
                  className="neighbor"
                >
                  <h4>
                    Tim Fischer
                  </h4>
                  <i
                    className="fas fa-angle-right"
                  />
                </div>,
                "className": "name-link",
                "replace": false,
                "to": "/neighbors/7",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <h4>
                      Tim Fischer
                    </h4>,
                    <i
                      className="fas fa-angle-right"
                    />,
                  ],
                  "className": "neighbor",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Tim Fischer",
                    },
                    "ref": null,
                    "rendered": "Tim Fischer",
                    "type": "h4",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": "fas fa-angle-right",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "i",
                  },
                ],
                "type": "div",
              },
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
